common/transport/amqp/test/_amqp_cbs_test.js

```diff
@@ -154,53 +196,91 @@ describe('ClaimsBasedSecurityAgent', function() {
+        assert(fakeRheaSession.open_receiver.calledOnce, 'open receiver NOT called once');
         cbs.forceDetach();
-        assert(fakeSender.forceDetach.calledOnce);
-        assert(fakeReceiver.forceDetach.calledOnce);
-        assert(fakeSender.detach.notCalled);
-        assert(fakeReceiver.detach.notCalled);
+        assert(amqpSenderForceDetachSpy.calledOnce);
+        assert(amqpReceiverForceDetachSpy.calledOnce);
+        assert(cbs._senderLink.detach.notCalled);
+        assert(cbs._senderLink.detach.notCalled);
         testCallback();
       });
     });
 
     /*Tests_SRS_NODE_AMQP_CBS_16_022: [The `forceDetach()` method shall call `forceDetach()` on all attached links.]*/
     it('forcefully detach attached links if called while attaching', function (testCallback) {
-      var fakeAmqpClient = new EventEmitter();
-      var fakeSender = new EventEmitter();
-      fakeSender.detach = sinon.stub().resolves();
-      fakeSender.forceDetach = sinon.stub();
-      var fakeReceiver = new EventEmitter();
-      fakeReceiver.detach = sinon.stub().resolves();
-      fakeReceiver.forceDetach = sinon.stub();
-      fakeAmqpClient.createSender = sinon.stub().resolves(fakeSender);
-      fakeAmqpClient.createReceiver = sinon.stub().resolves(fakeReceiver);
-      sinon.stub(fakeReceiver, "on").callsFake(function (eventName) {
+      var fakeRheaSession = new EventEmitter();
+
+      var cbs = new CBS(fakeRheaSession);
+
+      var fakeRheaReceiverLink = new EventEmitter();
+      var fakeRheaSenderLink = new EventEmitter();
+      var fakeSenderContext = {sender: fakeRheaSenderLink};
+      var fakeReceiverContext = {receiver: fakeRheaReceiverLink};
+      fakeRheaSession.open_sender = () => {};
+      sinon.stub(fakeRheaSession, 'open_sender').callsFake(() => {
+        process.nextTick( () => {
+          fakeRheaSenderLink.emit('sender_open', fakeSenderContext);
+        });
+        return fakeRheaSenderLink;
+      });
+      fakeRheaReceiverLink.remove = sinon.stub();
+      fakeRheaSenderLink.remove = sinon.stub();
+      fakeRheaSession.open_receiver = () => {};
+      sinon.stub(fakeRheaSession, 'open_receiver').callsFake(() => {
+        process.nextTick( () => {
+          fakeRheaReceiverLink.emit('receiver_open', fakeReceiverContext);
+        });
+        return fakeRheaReceiverLink;
+      });
+      var amqpSenderForceDetachSpy = sinon.spy(cbs._senderLink, 'forceDetach');
+      var amqpReceiverForceDetachSpy = sinon.spy(cbs._receiverLink, 'forceDetach');
+      cbs._senderLink.detach = sinon.stub();
+      cbs._receiverLink.detach = sinon.stub();
+
+      sinon.stub(fakeRheaReceiverLink, "on").callsFake(function (eventName) {
         if (eventName === 'message') {
           // sender is attached, receiver is attaching since we're registering for messages. good time to trigger a fake forceDetach
+          assert(fakeRheaSession.open_sender.calledOnce, 'open sender NOT called once');
```

Comment:
>'open sender NOT called once' [](start = 57, length = 29)

nit: this is _very_ obvious from the first half of the line. we could make it even more obvious by using a different test tool/syntax if you want. those messages feel redundant

Reply:
Hmm, I was feeling guilty that all of my asserts DON'T have the textual comment.  I thought that they were mostly there to help the reporting when the assert fails.  I thought that otherwise the error report from the unit test run is a bit 'terse' and not particularly helpful when an actual failure occurs.

---
In reply to: [209383243](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209383243) [](ancestors = 209383243)